web/src/TargetIpsPopup.jsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useState } from "react";
+import { useInstallerClient } from "./context/installer";
+import Popup from "./Popup";
+import { Button, Text } from "@patternfly/react-core";
+
+const initIpData = {
+  addresses: [],
+  hostname: ""
+};
+
+function formatIp(address, prefix) {
+  return address + "/" + prefix;
+}
+
+export default function TargetIpsPopup() {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const open = () => setIsOpen(true);
+  const close = () => setIsOpen(false);
+  const client = useInstallerClient();
+  const [state, setState] = useState(initIpData);
+  const ips = state.addresses.map((addr) => formatIp(addr.address, addr.prefix));
+  const firstIp = ips.length > 0 ? ips[0] : "";
+
+  useEffect(() => {
```

Comment:
You do not need to track whether the component is mounted. And sure you do not need an `async` function here. There might be better ways to do it, but I would change the approach to:

1. Extend `NetworkClient` to listen for changes in NetworkManager. I would expect to be possible to listen for a path or a signal to detect such changes.
2. Allow registering a callback for 1). Let's call it `onChange`.
3. Register the callback in the `useEffect`.

```javascript
useEffect(() => {
  return client.network.onChange(changes => {
    // you can call the setState here.
  });
}, []); // the dependencies array is empty so this useEffect executes only once
```

As the dependency array is empty, React will `client.network.onChange` only once, to register the callback. Beware that the `onChange` function should return a function to unsubscribe (see [onObjectChange as example](https://github.com/yast/d-installer/blob/98b9d2ea8344b8dafae0be8e16c01611db48081b/web/src/client/mixins.js#L63)). React will call that function when the component is unmounted, so no more *Can't perform a React state update... on an unmounted component* errors.

If listening for changes in NetworkManager is too hard, we could ignore the changes by now.